### ISSUE TITLE
AWS Credentials leaked when running Redshift Copy commands

### ISSUE THREAD

**Issue by jbeemster** [getredash/redash#2258]
### Issue Summary

When performing AWS copy commands from redash the _full_ query is stored to the events tables and emitted via the Webhook functionality.  If the user is not using IAM roles the AWS Access & Secret keys are stored into the events table.

A way to sanitise this input after submission so credentials are not stored would be great to prevent leaking credentials inadvertently.

### Steps to Reproduce

1. Load data via the COPY command using the Redash SQL editor

```
copy scratch.users from 's3://fake-bucket/seed/allusers_pipe.txt' 
access_key_id 'xxx'
secret_access_key 'xxx'
delimiter '|' region 'eu-west-1';
```

2. Check the output of the Webhook event

```
{
    "data": {
        "user_id": 1,
        "created_at": "2018-01-25T14:24:56",
        "object_type": "data_source",
        "org_id": 1,
        "object_id": 1,
        "additional_properties": {
            "ip": "xx.xx.xx.xx",
            "user_name": "Joshua Beemster",
            "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36",
            "query": "copy scratch.users from 's3://fake-bucket/seed/allusers_pipe.txt' \naccess_key_id 'xxx'\nsecret_access_key 'xxx'\ndelimiter '|' region 'eu-west-1';"
        },
        "action": "execute_query"
    },
    "schema": "iglu:io.redash.webhooks/event/jsonschema/1-0-0"
}
```

3. Check the row stored in the events table

```
{"ip": "10.0.30.34", "user_name": "Joshua Beemster", "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36", "query": "copy scratch.users from 's3://fake-bucket/seed/allusers_pipe.txt' \naccess_key_id 'xxx'\nsecret_access_key 'xxx'\ndelimiter '|' region 'eu-west-1';"}
```

### Technical details:

* Redash Version: 3.0.0+b3134

**Comment by tonyjiangh** [comment]
Honestly, I don't think it's a good idea to issue copy command through re:dash at all.

If you do care about security and data safety, it's even better not to use any user with privilege more than 'select' at datasource setting. It would reduce the possibility of misoperations that damage the data.

Though, the above is based on the perception that re:dash is aiming for connect, visualize and share data.
If we are going to expand the scope to data management, than this request is more than reasonable.

**Comment by jbeemster** [comment]
> Honestly, I don't think it's a good idea to issue copy command through re:dash at all.

I agree with you on this point!  The point here however is more that re:dash stores the full query irrespective of what has been passed into it.  The COPY command is just one example of how credentials could be leaked.

The ability to actively protect users from inadvertently leaking secrets can, in my opinion, only be a good thing!  A function for cleansing secrets before they are emitted via the Webhook or stored to the events table would ensure that re:dash is safe to use for ever more use-cases than the read-only / data visualisation option that you have highlighted.

**Comment by tonyjiangh** [comment]
> The ability to actively protect users from inadvertently leaking secrets can, in my opinion, only be a good thing!

You're right.
